src/check-dsl.ts

```diff
@@ -168,15 +184,34 @@ export const basicValidateRelation = (
   });
 };
 
-export const checkDSL = (codeInEditor: string) => {
+export const checkDSL = (
+  codeInEditor: string,
+  typeValidation: string = defaultTypeRule,
```

Comment:
Can we pass these in options? As we add more things, the list will grow

Reply:
We should validate that what is passed can be converted to a regex and throw an appropriate error if not

Reply:
Done